pkg/agent/agent_test.go

```diff
@@ -708,3 +712,261 @@ func TestRestorePortConfigs(t *testing.T) {
+		},
+		{
+			name: "datapath-id exists in other_config on OVS bridge",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(map[string]string{
+					ovsconfig.OVSOtherConfigDatapathIDKey: "datapathId",
+				}, nil)
+			},
+		},
+		{
+			name: "generate and set datapath ID for OVS bridge",
+			expectedCalls: func(m *ovsconfigtest.MockOVSBridgeClient) {
+				m.EXPECT().GetOVSOtherConfig().Return(map[string]string{}, nil)
+				m.EXPECT().SetDatapathID(mock.Any())
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			controller := mock.NewController(t)
+			mockOVSBridgeClient := ovsconfigtest.NewMockOVSBridgeClient(controller)
+			initializer := newAgentInitializer(mockOVSBridgeClient, nil)
+			tt.expectedCalls(mockOVSBridgeClient)
+
+			err := initializer.setOVSDatapath()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func mockIPsecPSKEnv(name string) func() {
+	os.Setenv(ipsecPSKEnvKey, name)
+	return func() { os.Unsetenv(ipsecPSKEnvKey) }
+}
+
+func TestReadIPSecPSK(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectedErr string
+	}{
+		{
+			name:        "IPsec PSK env variable not set",
+			expectedErr: "IPsec PSK environment variable 'ANTREA_IPSEC_PSK' is not set or is empty",
+		},
+		{
+			name: "IPsec PSK env variable set",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			initializer := &Initializer{
+				networkConfig: &config.NetworkConfig{
+					IPsecConfig: config.IPsecConfig{},
+				},
+			}
+			if tt.expectedErr == "" {
+				defer mockIPsecPSKEnv("key")()
+			}
+
+			err := initializer.readIPSecPSK()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGetNodeConfig(t *testing.T) {
+	nodeConfig := &config.NodeConfig{}
+	initializer := &Initializer{
+		nodeConfig: nodeConfig,
+	}
+	assert.Equal(t, nodeConfig, initializer.GetNodeConfig())
+}
+
+func TestWaitForIPSecMonitorDaemon(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectedErr string
+	}{
+		{
+			name:        "IPsec monitor is not running",
+			expectedErr: "IPsec was requested, but the OVS IPsec monitor does not seem to be running",
+		},
+		{
+			name: "IPsec monitor running",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			initializer := &Initializer{}
+			if tt.expectedErr == "" {
+				appFS := afero.NewMemMapFs()
+				err := appFS.MkdirAll("/var/run/openvswitch", 0777)
+				require.NoError(t, err)
+				_, err = appFS.Create("/var/run/openvswitch/ovs-monitor-ipsec.pid")
+				require.NoError(t, err)
+				defaultFs = appFS
+				defer func() {
+					defaultFs = afero.NewOsFs()
+				}()
+			}
+
+			err := initializer.waitForIPsecMonitorDaemon()
```

Comment:
Use fake clock, instead of letting the test really run 10s. The unit test would run a long time if every test wait their real timeout.
Seach `clock clockutils.Clock = clockutils.RealClock{}` in the repo for reference.

Reply:
you didn't mock clock, the test will still run 10s.

Reply:
done, PTAL. 
Thanks @tnqn